src/Conversion/ONNXToMhlo/Tensor/Transpose.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Transpose.cpp - Lowering Transpose Op ---------------===//
+//
+// Copyright 2019-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX Transpose Operator to Mhlo dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToMhlo/ONNXToMhloCommon.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+#include "src/Support/TypeUtilities.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+struct ONNXTransposeOpLoweringToMhlo : public ConversionPattern {
+  ONNXTransposeOpLoweringToMhlo(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXTransposeOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    ONNXTransposeOpAdaptor operandAdaptor(operands);
+    ONNXTransposeOp transposeOp = llvm::cast<ONNXTransposeOp>(op);
+    Location loc = op->getLoc();
+
+    // Operands
+    Value data = operandAdaptor.data();
+
+    // Convert the output type
+    Type outputType = *op->result_type_begin();
+    assert(outputType.isa<ShapedType>() && "Expected ShapedType");
+    ShapedType outputShapedType = outputType.cast<ShapedType>();
+    int64_t rank = outputShapedType.getShape().size();
+
+    // Attributes
+    llvm::Optional<ArrayAttr> permAttr = transposeOp.perm();
+    DenseIntElementsAttr permAxis;
+    RankedTensorType permAxisType =
+        RankedTensorType::get({rank}, rewriter.getI64Type());
+    SmallVector<int64_t, 4> permAxisList;
+    if (permAttr.hasValue()) {
```

Comment:
I think this should be `has_value()`, and thats what causes the windows pipeline to fail.

`hasValue()` was deprecated in MLIR

Reply:
Fixed, thanks @philass